packit/upstream.py

```diff
@@ -452,10 +452,20 @@ def create_archive(self, version: str = None) -> str:
             if not outputs:
                 raise PackitException("No output from create-archive action.")
             # one of the returned strings has to contain existing archive name
-            for archive_name in reversed(outputs):
-                if Path(archive_name.strip()).is_file():
-                    logger.info(f"Created archive: {archive_name.strip()}")
-                    return archive_name
+            for output in reversed(outputs):
+                for archive_name in reversed(output.splitlines()):
```

Comment:
Maybe something like this to flatten the for loop:

```python
for archive_name in reversed(
    [line for output in outputs for line in output.splitlines()]
):
    pass
```

or maybe this:

```python
for archive_name in reversed(sum([output.splitlines() for output in [""]], [])):
    pass
```

But maybe it is not well readable and needs to save all lines in the list -- feel free to ignore this comment. I just don't like having a lot of levels in the code.

Reply:
I'd prefer it the way it is now - it's written stupidly I agree, but IMO has the greatest potential to be understood quickly in the future :)